### PR TITLE
New UI - Modify the printer selector

### PR DESCRIPTION
**Pull request by diegopradogesto**
This PR adds the necessary work for the printer selector in the new Cura UI. The last commits makes reference also to the views selector in the preview stage.

CURA-5942

### CODE REVIEW

File: cura/PrinterOutputDevice.py [hunk 1]
```diff
@@ -211,6 +211,11 @@ def _updateUniqueConfigurations(self) -> None:
         self._unique_configurations.sort(key = lambda k: k.printerType)
         self.uniqueConfigurationsChanged.emit()
 
+    # Returns the unique configurations of the printers within this output device
+    @pyqtProperty("QVariantList", notify = uniqueConfigurationsChanged)
```
Comment (nallath) [inline cura/PrinterOutputDevice.py]:
```suggestion
    @pyqtProperty("QStringList", notify = uniqueConfigurationsChanged)
```

File: resources/qml/ExpandableComponent.qml [hunk 2]
```diff
@@ -10,6 +10,13 @@ import UM 1.2 as UM
 Item
 {
     id: base
+
+    // Enumeration with the different possible alignments of the popup with respect of the headerItem
+    enum PopupAlignment {
```
Comment (nallath) [inline resources/qml/ExpandableComponent.qml]:
```suggestion
    enum PopupAlignment 
    {
```

File: resources/qml/PrinterSelector/MachineSelector.qml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        font: UM.Theme.getFont("medium")
+        color: UM.Theme.getColor("text")
+        iconSize: UM.Theme.getSize("machine_selector_icon").width
+
+        UM.RecolorImage
+        {
+            id: icon
+
+            anchors.bottom: parent.bottom
+            x: UM.Theme.getSize("thick_margin").width
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelector.qml]: Why use x and not an anchor?

File: resources/qml/PrinterSelector/MachineSelector.qml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            color: UM.Theme.getColor("primary")
+            visible: isNetworkPrinter && isPrinterConnected
+
+            // Make a themable circle in the background so we can change it in other themes
+            Rectangle
+            {
+                id: iconBackground
+                anchors.centerIn: parent
+                // Make it a bit bigger so there is an outline
+                width: parent.width + 2
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelector.qml]: Shouldn't the +2 be a certain margin size? Like the lining?

File: resources/qml/PrinterSelector/MachineSelectorButton.qml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import UM 1.1 as UM
+import Cura 1.0 as Cura
+
+Button
+{
+    id: machineSelectorButton
+
+    width: parent.width
+    height: UM.Theme.getSize("action_button").height
+    leftPadding: Math.round(1.5 * UM.Theme.getSize("default_margin").width)
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorButton.qml]: Soo thick margin?

File: resources/qml/PrinterSelector/MachineSelectorButton.qml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    rightPadding: Math.round(1.5 * UM.Theme.getSize("default_margin").width)
+    checkable: true
+
+    property var outputDevice: null
+    property var printerTypesList: []
+
+    function setPrinterTypesList()
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorButton.qml]:
```suggestion
    function updatePrinterTypesList()
```
We tend to use set for actual setters, but that's not what this function does.

File: resources/qml/PrinterSelector/MachineSelectorButton.qml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        id: mouseArea
+        anchors.fill: parent
+        onPressed: mouse.accepted = false
+        hoverEnabled: true
+    }
+
+    Connections
+    {
+        target: outputDevice
+        onUniqueConfigurationsChanged: setPrinterTypesList()
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorButton.qml]:
```suggestion
        onUniqueConfigurationsChanged: updatePrinterTypesList()
```

File: resources/qml/PrinterSelector/MachineSelectorButton.qml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    Connections
+    {
+        target: Cura.MachineManager
+        onOutputDevicesChanged: setPrinterTypesList()
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorButton.qml]:
```suggestion
        onOutputDevicesChanged: updatePrinterTypesList()
```

File: resources/qml/PrinterSelector/MachineSelectorButton.qml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    Component.onCompleted: setPrinterTypesList()
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorButton.qml]:
```suggestion
    Component.onCompleted: updatePrinterTypesList()
```

File: resources/qml/PrinterSelector/MachineSelectorList.qml [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    Repeater
+    {
+        id: networkedPrinters
+
+        model: UM.ContainerStacksModel
+        {
+            id: networkedPrintersModel
+            filter: {"type": "machine", "um_network_key": "*", "hidden": "False"}
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorList.qml]: { needs to be on newline.

File: resources/qml/PrinterSelector/MachineSelectorList.qml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    Repeater
+    {
+        id: virtualPrinters
+
+        model: UM.ContainerStacksModel
+        {
+            id: virtualPrintersModel
+            filter: {"type": "machine", "um_network_key": null}
```
Comment (nallath) [inline resources/qml/PrinterSelector/MachineSelectorList.qml]: { needs to be on newline

File: resources/qml/PrinterSelector/MachineSelector.qml (hunk 3 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelector.qml]: Done in 5b940b524204ca56b5fafe05bad28d5e477b8ea7

File: resources/qml/PrinterSelector/MachineSelectorButton.qml (hunk 5 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in 5b940b524204ca56b5fafe05bad28d5e477b8ea7

Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in 5b940b524204ca56b5fafe05bad28d5e477b8ea7

File: resources/qml/PrinterSelector/MachineSelectorButton.qml (hunk 9 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in fa1ef5c45c70ebdebaef3d9d19b026e6586523b2

File: resources/qml/PrinterSelector/MachineSelectorButton.qml (hunk 8 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in fa1ef5c45c70ebdebaef3d9d19b026e6586523b2

File: resources/qml/PrinterSelector/MachineSelectorButton.qml (hunk 7 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in fa1ef5c45c70ebdebaef3d9d19b026e6586523b2

File: resources/qml/PrinterSelector/MachineSelectorButton.qml (hunk 6 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorButton.qml]: Done in fa1ef5c45c70ebdebaef3d9d19b026e6586523b2

File: resources/qml/PrinterSelector/MachineSelector.qml (hunk 4 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelector.qml]: Done in 63fab9f038e8615432a8c592138e915351131386

File: resources/qml/PrinterSelector/MachineSelectorList.qml (hunk 10 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorList.qml]: Done in 21bfa6e4c20819e09b8cb2f27f62427dc71c8476

File: resources/qml/PrinterSelector/MachineSelectorList.qml (hunk 11 above)
Comment (diegopradogesto) [reply resources/qml/PrinterSelector/MachineSelectorList.qml]: Done in 21bfa6e4c20819e09b8cb2f27f62427dc71c8476

**Pull request merged**